src/openrct2/CMakeLists.txt

```diff
@@ -17,6 +17,11 @@ add_library(${PROJECT_NAME} ${OPENRCT2_CORE_SOURCES} ${OPENRCT2_CORE_MM_SOURCES}
 set_target_properties(${PROJECT_NAME} PROPERTIES PREFIX "")
 SET_CHECK_CXX_FLAGS(${PROJECT_NAME})
 
+# GCC likes us to pass the -lstdc++fs flag to link C++17 filesystem implementation.
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT MINGW)
```

Comment:
This is also sometimes needed when Clang is used on linux. By default clang is using libstdc++ (the same as GCC) on linux and it also needs this flag. In CMake, there seems not to be an easy way to check what standard library is being used. I usually put this condition as `if(UNIT NOT APPLE)`.